book/python/api.md

```diff
@@ -32,111 +30,163 @@ The recommended way to create your topology structure is by using the [Applicati
+You create complete pipeline by terminating a partial pipeline with a call to `to_sink` or `to_sinks`. For example:
 
-The `ApplicationBuilder` class in `wallaroo` is a utility for constructing application topologies. It also provides some additional validation on the input parameters.
+```python
+inputs = wallaroo.source("Source Name", source_config)
+complete_pipeline = (inputs
+    .to(my_computation)
+    .to_sink(sink_config))
+```
 
-#### `ApplicationBuilder` Methods
+Our pipeline is now complete:
 
-##### `__init__(name)`
+```
+Source -> my_computation -> Sink
+```
 
-Create a new application with the name `name`.
+Unless a call to `to` using a stateless computation is preceded by a call to `key_by` (which partitions messages by key), there are no guarantees around the order messages will be processed in. That's because Wallaroo might parallelize a stateless computation if that is beneficial for scaling. That means the execution graph for the above pipeline could look like this:
 
-##### `new_pipeline(name, source_config)`
+```
+         /-> my_stateless_computation -\
+        /                               \
+Source ----> my_stateless_computation ----> Sink
+        \                               /
+         \-> my_stateless_computation -/
+```
 
-Create a new pipeline.
+Some messages will be routed to each of the parallel computation instances. When they merge again at the sink, these messages will be interleaved in a non-deterministic fashion.
 
-`name` must be a string.
+#### Merging Partial Pipelines
 
-`source_config` must be one of the [SourceConfig](#source) classes.
+You can merge two partial pipelines to form a new partial pipline. For example:
 
-If you're adding more than one pipeline, make sure to call `done()` before creating another pipeline.
+```python
+inputs1 = wallaroo.source("Source 1", source_config)
+partial_pipeline1 = inputs1.to(computation1)
 
-##### `done()`
+inputs2 = wallaroo.source("Source 2", source_config)
+partial_pipeline2 = inputs2.to(computation2)
 
-Close the current pipeline object.
+partial_pipeline = inputs1.merge(inputs2)
+```
 
-This is necessary if you intend to add another pipeline.
+The resulting partial pipeline could be 
 
-##### `to(computation)`
+```
+Source1 -> computation1 ->\
+                           \
+                            ->
+                           /
+Source2 -> computation2 ->/
+```
+
+Again, the hanging arrow indicates we can still add more stages, and that to complete the pipeline we still need one or more sinks. You could also merge this partial pipeline with yet more partial pipelines. When you merge partial pipelines in this way, you are not creating a join in the sense familiar from SQL joins. Instead, you are combining two streams into one, with messages from the first stream interwoven with messages from the second. That combined stream is then passed to the next stage following the hanging arrow.
```

Comment:
"You could also merge this partial pipeline with yet more partial pipelines" maybe "another" or "additional" partial pipelines? "Yet more" reads a little awkwardly to me.